### ISSUE TITLE
Exited(255) Unhandled exception: SocketException: Write failed (OS Error: Broken pipe, errno = 32), address = 127.0.0.1, port = 63467

### ISSUE THREAD

**Issue by Caesar0** [ra1u/redis-dart#49]
Occurs when the redis service shuts down unexpectedly!

=====================================================================================
```dart
Unhandled exception:
SocketException: Write failed (OS Error: Broken pipe, errno = 32), address = 127.0.0.1, port = 63467
#0      _NativeSocket.write (dart:io-patch/socket_patch.dart:1119:34)
#1      _RawSocket.write (dart:io-patch/socket_patch.dart:1765:15)
#2      _Socket._write (dart:io-patch/socket_patch.dart:2198:18)
#3      _SocketStreamConsumer.write (dart:io-patch/socket_patch.dart:1946:26)
#4      _SocketStreamConsumer.addStream.<anonymous closure> (dart:io-patch/socket_patch.dart:1920:11)
#5      _rootRunUnary (dart:async/zone.dart:1444:13)
#6      _RootZone.runUnaryGuarded (dart:async/zone.dart:1623:7)
#7      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
#8      _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271:7)
#9      _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:733:19)
#10     _StreamController._add (dart:async/stream_controller.dart:607:7)
#11     _StreamController.add (dart:async/stream_controller.dart:554:5)
#12     _StreamSinkImpl.add (dart:io/io_sink.dart:136:17)
#13     _Socket.add (dart:io-patch/socket_patch.dart:2045:38)
#14     RedisConnection._sendraw
package:redis/connection.dart:75
#15     Command.send_object
package:redis/command.dart:27
#16     Cas.watch.<anonymous closure>
package:redis/cas.dart:31
#17     Future.doWhile.<anonymous closure> (dart:async/future.dart:535:26)
#18     _rootRunUnary (dart:async/zone.dart:1436:47)
#19     _CustomZone.runUnary (dart:async/zone.dart:1335:19)
#20     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
#21     _CustomZone.bindUnaryCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1281:26)
#22     Future.doWhile (dart:async/future.dart:551:18)
#23     Cas.watch
package:redis/cas.dart:29


```

**Comment by ra1u** [comment]
I think that you need are missing appropriate error catching.

What you receive is exception of type SocketException that is result of disconnect. 

What do you to expect from this redis-dart library when redis service shuts down unexpectedly and socket is shut down?

**Comment by ra1u** [comment]
There is simple patch available for dart.
Let see how fast we can get this trough.

https://dart-review.googlesource.com/c/sdk/+/218700

**Comment by ra1u** [comment]
We have got some response from Dart and there is workaround. One needs to close socket just when error is received. 

This [workaround](https://github.com/ra1u/redis-dart/commit/053dc90d47d34bbae6d28c98a5701df8aba9b1a3) is now implemented in master branch  and will be released in next days.

Please test and report if you have issues.

**Comment by ra1u** [comment]
Fixed and released whit v2.2.0.

Thank you for your input.

**Comment by Caesar0** [comment]
Thanks sir,I tested it and it has been repaired.
Thank you for solving this problem in your busy schedule.
